homeassistant/components/lektrico/config_flow.py

```diff
@@ -0,0 +1,137 @@
+"""Config flow for Lektrico Charging Station."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from lektricowifi import Device, DeviceConnectionError
+import voluptuous as vol
+
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import (
+    ATTR_HW_VERSION,
+    ATTR_SERIAL_NUMBER,
+    CONF_HOST,
+    CONF_TYPE,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.httpx_client import get_async_client
+
+from .const import DOMAIN
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+
+class LektricoFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a Lektrico config flow."""
+
+    VERSION = 1
+
+    _host: str
+    _name: str
+    _serial_number: str
+    _board_revision: str
+    _device_type: str
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return self._async_show_setup_form()
+
+        self._host = user_input[CONF_HOST]
+
+        # obtain serial number
+        try:
+            await self._get_lektrico_device_settings_and_treat_unique_id()
+        except DeviceConnectionError:
+            return self._async_show_setup_form(errors={CONF_HOST: "cannot_connect"})
+
+        return self._async_create_entry()
+
+    @callback
+    def _async_show_setup_form(
+        self,
+        user_input: dict[str, Any] | None = None,
+        errors: dict[str, str] | None = None,
+    ) -> ConfigFlowResult:
+        """Show the setup form to the user."""
+        if user_input is None:
+            user_input = {}
+
+        schema = self.add_suggested_values_to_schema(STEP_USER_DATA_SCHEMA, user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=schema,
+            errors=errors or {},
+        )
+
+    @callback
+    def _async_create_entry(self) -> ConfigFlowResult:
+        return self.async_create_entry(
+            title=self._name,
+            data={
+                CONF_HOST: self._host,
+                ATTR_SERIAL_NUMBER: self._serial_number,
+                CONF_TYPE: self._device_type,
+                ATTR_HW_VERSION: self._board_revision,
+            },
+        )
```

Comment:
These can be combined imo

You can structure the user step to `if user_input is not None:` and then you can let everything overflow in the show forn

Reply:
I made the changes, thanks.